main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -1808,17 +1808,23 @@ object Weeder2 {
 
     private def visitInvokeConstructorExpr(tree: Tree)(implicit sctx: SharedContext): Validation[Expr, CompilationMessage] = {
       expect(tree, TreeKind.Expr.InvokeConstructor)
-      mapN(Types.pickType(tree), pickRawArguments(tree, synctx = SyntacticContext.Expr.New)) {
-        (tpe, exps) =>
-          tpe match {
-            case WeededAst.Type.Ambiguous(qname, _) if qname.isUnqualified =>
-              Expr.InvokeConstructor(qname.ident, exps, tree.loc)
-            case _ =>
-              val error = IllegalQualifiedName(tree.loc)
-              sctx.errors.add(error)
-              Expr.Error(error)
+      val expsValidation = tryPick(TreeKind.ArgumentList, tree) match {
```

Comment:
I dont like this akward pattern match.

Just inline it, i.e. 

```
mapN(Types.pickType(tree), tryPick(TreeKind.ArgumentList, tree)) match { ... }
```

and do the three cases in `match {. .. }`

Reply:
I see. Let me think

Reply:
OK its fine